File: src/test/kotlin/io/ashdavies/rx/rxtasks/RxTasksTest.java
package io.ashdavies.rx.rxtasks;

import com.google.android.gms.tasks.Task;
import com.google.android.gms.tasks.Tasks;
import org.junit.Test;

public class RxTasksTest {

  @Test
  public void shouldRepresentTaskAsCompletable() {
    Task<Void> task = Tasks.forResult(null);

    RxTasks
        .completable(task)
        .test()
        .assertResult();
  }

  @Test
  public void shouldRepresentTaskAsSingle() {
    Task<Integer> task = Tasks.forResult(42);

    RxTasks
        .single(task)
        .test()
        .assertResult(42);
  }
}

File: library/src/test/kotlin/io/ashdavies/rx/rxtasks/RxTasksTest.kt
package io.ashdavies.rx.rxtasks

import com.google.android.gms.tasks.Task
import com.nhaarman.mockito_kotlin.mock
import com.nhaarman.mockito_kotlin.then
import org.junit.Ignore
import org.junit.Test
import java.util.concurrent.Executor

internal class RxTasksTest {

  private val executor = mock<Executor>()

  @Test
  fun `should delegate to completable extension`() {
    val task = mock<Task<Void>>()

    RxTasks.completable(task, executor)

    then(task).should().toCompletable(executor)
  }

  @Test
  @Ignore
  fun `should delegate to single extension`() {
    val task = mock<Task<Boolean>>()

    RxTasks.single(task, executor)

    then(task).should().toSingle(executor)
  }
}
